lib/Alchemy/Phrasea/Controller/Api/V3/V3MonitorDataController.php

```diff
@@ -169,15 +100,218 @@ public function indexAction(Request $request)
+
+        $ret = [
+            'unit' => $divider === 1 ? $unit : ucfirst($unit), // octet => octet ; mo => Mo
+            'databox' => []
+        ];
+
+        $databox = $this->findDataboxById($databoxId);
+
+        // get volumes by db
+
+        $stmt = $databox->get_connection()->prepare($sqlByDb);
+        $stmt->execute();
+        $row = $stmt->fetch(PDO::FETCH_ASSOC);
+        $stmt->closeCursor();
+
+        $ret['databox'] = [
+            'sbas_id'   => $databox->get_sbas_id(),
+            'viewname'  => $databox->get_viewname(),
+            'count'     => (int)$row['n'],
+            'size'      => round($row['size'], 2),
+            'disksize'  => round($row['disksize'], 2)
+        ];
+
+        if ($getDetails) {
+            list($collections, $subdefs) = $this->getVolumeDetails($databox, $sqlByColl, $sqlByName);
+
+            $ret['databox']['collections']  = $collections;
+            $ret['databox']['subdefs']      = $subdefs;
+        }
+
+        // get volumes of downloads
+
+        $sql = "SELECT `data` FROM `Tokens` WHERE `type`='download'";
+        $stmt = $this->getApplicationBox()->get_connection()->prepare($sql);
+        $stmt->execute();
+        $size = 0;
+        $disksize = 0;
+        $n = 0;
+        while ($row = $stmt->fetch(PDO::FETCH_ASSOC)) {
+            try {
+                $found = false;
+                $data = unserialize($row['data']);
+                foreach ($data['files'] as $file) {
+                    // get only for the needed databoxId
+                    if ($file['databox_id'] == $databoxId) {
+                        $found = true;
+                        foreach ($file['subdefs'] as $subdef) {
+                            $size += $subdef['size'];
+                            $disksize += ceil($subdef['size'] / $blocksize) * $blocksize;
+                        }
+                    }
+                }
+
+                if ($found) {
+                    $n++;
+                }
+            }
+            catch (\Exception $e) {
+                // ignore
+            }
+        }
+
+        $stmt->closeCursor();
+
+        $ret['downloads'] = [
+            'sbas_id'   => $databoxId,
+            'count'     => $n,
+            'size'      => round($size / $divider, 2),
+            'disksize'  => round($disksize / $divider, 2)
         ];
 
+        // get lazaret volume for the databox
+
+        $sql = "SELECT count(*) AS n , SUM(`size`) AS size ".
+            " FROM `LazaretFiles` AS L ".
+            " LEFT JOIN `bas` AS b ON L.`base_id`=b.`base_id`".
+            " WHERE L.`size` IS NOT NULL AND b.`sbas_id`=". $databoxId;
+
+
+        $stmt = $this->getApplicationBox()->get_connection()->prepare($sql);
+        $stmt->execute();
+
+        $row = $stmt->fetch(PDO::FETCH_ASSOC);
+        $stmt->closeCursor();
+
+        $disksize = ceil($row['size'] / $blocksize) * $blocksize;;
```

Comment:
NO...
disksize is **per file**, eg. if blocksize=16ko, the disksize of every file is a multiple of 16ko :
- 10 small files (1 ko) = 10 * 16 ko ==> 160 ko (good)

here we get 10 * 1 ko = 10 ko ==> 16 ko (wrong)

Do blocksize, divider etc **in sql**, like in other requests.

Reply:
ok